src/common/meta/src/error.rs

```diff
@@ -741,6 +768,8 @@ impl ErrorExt for Error {
             | ParseNum { .. }
             | InvalidRole { .. }
             | EmptyDdlTasks { .. } => StatusCode::InvalidArguments,
+
+            ConnectPostgres { .. } => StatusCode::IllegalState,
```

Comment:
Seems to be `StatusCode::InvalidArguments`?

Reply:
The database could fail too i think IllegalState make sense
 https://github.com/GreptimeTeam/greptimedb/pull/4421#discussion_r1704801267